modules/nexverseBidAdapter.js

```diff
@@ -0,0 +1,447 @@
+      logWarn(`${LOG_WARN_PREFIX} No valid bids in the response.`);
+      return bidResponses;
+    }
+
+    response.seatbid.forEach((seatbid) => {
+      seatbid.bid.forEach((bid) => {
+        const bidResponse = {
+          requestId: bid.impid,
+          cpm: bid.price,
+          currency: response.cur || 'USD',
+          width: bid.w || 0,
+          height: bid.h || 0,
+          creativeId: bid.crid || bid.id,
+          ttl: 300,
+          netRevenue: true,
+          meta: {},
+        };
+
+        // Determine media type and assign the ad content
+        if (bid.ext && bid.ext.mediaType) {
+          bidResponse.mediaType = bid.ext.mediaType;
+        } else if (bid.adm && bid.adm.indexOf('<VAST') !== -1) {
+          bidResponse.mediaType = VIDEO;
+          bidResponse.vastXml = bid.adm;
+        } else if (bid.adm && bid.adm.indexOf('"native"') !== -1) {
+          bidResponse.mediaType = NATIVE;
+          bidResponse.native = parseNativeResponse(bid.adm);
+        } else {
+          bidResponse.mediaType = BANNER;
+          bidResponse.ad = bid.adm || '';
+        }
+
+        // Handle advertiser domains
+        if (bid.adomain && isArray(bid.adomain)) {
+          bidResponse.meta.advertiserDomains = bid.adomain;
+        } else {
+          bidResponse.meta.advertiserDomains = [];
+        }
+
+        // Include 'nurl' if provided
+        if (bid.nurl) {
+          bidResponse.nurl = bid.nurl;
+        }
+
+        bidResponses.push(bidResponse);
+      });
+    });
+
+    return bidResponses;
+  },
+
+  /**
+   * Registers user sync pixels.
+   *
+   * @param {Object} syncOptions - Configuration specifying which user syncs are allowed.
+   * @param {Array} serverResponses - Array of server responses.
+   * @param {Object} gdprConsent - GDPR consent information.
+   * @param {string} uspConsent - US Privacy consent string.
+   * @returns {Array} Array of user syncs to be executed.
+   */
+  getUserSyncs(syncOptions, serverResponses, gdprConsent, uspConsent) {
+    const syncs = [];
+
+    if (syncOptions.iframeEnabled) {
+      let syncUrl = `${ENDPOINT_URL}/sync`;
+
+      const params = [];
+
+      // GDPR
+      if (gdprConsent) {
+        params.push(`gdpr=${gdprConsent.gdprApplies ? 1 : 0}`);
+        params.push(`gdpr_consent=${encodeURIComponent(gdprConsent.consentString || '')}`);
+      }
+
+      // CCPA
+      if (uspConsent) {
+        params.push(`us_privacy=${encodeURIComponent(uspConsent)}`);
+      }
+
+      if (params.length > 0) {
+        syncUrl += '?' + params.join('&');
+      }
+
+      syncs.push({
+        type: 'iframe',
+        url: syncUrl,
+      });
+    }
+
+    return syncs;
+  },
+};
+
+/**
+ * Builds the OpenRTB 2.5 request payload.
+ *
+ * @param {Object} bid - The bid request object.
+ * @param {Object} bidderRequest - The bidder request object.
+ * @returns {Object|null} The OpenRTB 2.5 request payload or null if missing mandatory parameters.
+ */
```

Comment:
Why not use the utility function available for this

Reply:
Done.